src/lib/utils/thorchain/index.ts

```diff
@@ -186,6 +186,7 @@ export const getThorchainFromAddress = async ({
 
     const chainAdapter = getChainAdapterManager().get(chainId)!
 
+    // @ts-ignore this is one of the rare cases where we want to actually bypass on-device derivation, but for Ledger only
```

Comment:
We could avoid this ts-ignore and maintain type safety by doing something like:

```ts
const firstReceiveAddress =
      isLedger(wallet) && accountId
        ? await chainAdapter.getAddress({
            wallet,
            accountNumber: bip44Params.accountNumber,
            accountType,
            index: 0,
            pubKey: fromAccountId(accountId).account,
          })
        : await chainAdapter.getAddress({
            wallet,
            accountNumber: bip44Params.accountNumber,
            accountType,
            index: 0,
            checkLedgerAppOpenIfLedgerConnected: () => Promise.resolve(),
          })
```

Reply:
Ohhh I like the `checkLedgerAppOpenIfLedgerConnected: () => Promise.resolve(),` idea @0xApotheosis! Keeps the type-safety in for most cases, but allows us to pass a noop in those edge cases 🧠 that may be a much better contender than lying to the type system/opt-chaining to an empty string!

Reply:
e08c8e4ce8